pkg/cloudprovider/metrics/cloudprovider.go

```diff
@@ -68,25 +87,66 @@ func Decorate(cloudProvider cloudprovider.CloudProvider) cloudprovider.CloudProv
 
 func (d *decorator) Create(ctx context.Context, machine *v1alpha5.Machine) (*v1alpha5.Machine, error) {
 	defer metrics.Measure(methodDurationHistogramVec.WithLabelValues(injection.GetControllerName(ctx), "Create", d.Name()))()
-	return d.CloudProvider.Create(ctx, machine)
+	machine, err := d.CloudProvider.Create(ctx, machine)
+	if err != nil {
+		errorsTotalCounter.With(prometheus.Labels{
+			metricLabelError: getErrorTypeLabelValue(err),
+		}).Inc()
+	}
+	return machine, nil
```

Comment:
We still need to propagate the err for the handling logic to work correctly. 
```suggestion
	return machine, err
```

Reply:
done